pkg/gotenberg/pdfengine.go

```diff
@@ -113,6 +117,12 @@ type PdfEngine interface {
 
 	// WriteMetadata writes the metadata into a given PDF file.
 	WriteMetadata(ctx context.Context, logger *zap.Logger, metadata map[string]interface{}, inputPath string) error
+
+	// ProtectWithPassword adds password protection to a PDF file.
+	// The userPassword is required to open the document.
+	// The ownerPassword provides full access to the document.
+	// If the ownerPassword is empty, it defaults to the userPassword.
+	ProtectWithPassword(ctx context.Context, logger *zap.Logger, inputPath, outputPath string, userPassword, ownerPassword string) error
```

Comment:
```suggestion
	AddPassword(ctx context.Context, logger *zap.Logger, inputPath, outputPath string, userPassword, ownerPassword string) error
```

To reflect the route naming?

Reply:
I am wondering if the route name should be encrypt instead of `password/add` and later on `password/remove`. What do you think @gulien?

Reply:
Good idea! Let's rename it to `Encrypt`!